source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -527,6 +527,48 @@ void Filter::sendTrailers(ProcessorState& state, const Http::HeaderMap& trailers
   stats_.stream_msgs_sent_.inc();
 }
 
+void Filter::onNewTimeout(
+    const std::unique_ptr<envoy::service::ext_proc::v3::ProcessingResponse>&& response) {
+  // In the following conditions the function just bails out and the message is ignored.
+  // 1) The message_timeout field is not set, or even it is set but < 1ms;
+  // 2) or the traffic_direction field is not set to be INBOUND or OUTBOUND.
+  const auto new_timeout = response->new_timeout();
+  const auto message_timeout_ms_opt = PROTOBUF_GET_OPTIONAL_MS(new_timeout, message_timeout);
+  if (!message_timeout_ms_opt) {
+    ENVOY_LOG(warn, "Ext_proc server new timeout setting is missing. Ignoring the message.");
+    return;
+  }
+  const auto message_timeout_ms = message_timeout_ms_opt->count();
+  const uint32_t min_timeout_ms = 1;
+  const uint32_t max_timeout_ms = 10000;
+  if (message_timeout_ms < min_timeout_ms || message_timeout_ms > max_timeout_ms) {
+    ENVOY_LOG(warn, "Ext_proc server new timeout setting is out of range. "
+                    "Ignoring the message.");
+    return;
+  }
+  const auto traffic_direction = new_timeout.traffic_direction();
+  const bool inbound_traffic =
+      (traffic_direction == envoy::config::core::v3::TrafficDirection::INBOUND);
+  const bool outbound_traffic =
+      (traffic_direction == envoy::config::core::v3::TrafficDirection::OUTBOUND);
+  if (!inbound_traffic && !outbound_traffic) {
```

Comment:
maybe just traffic_direction == unspecified

Reply:
Thanks for the comments!  There are a couple of concerns to do that way:
1) New elements can be added to the enum in the future.
2) Somebody can send a direction like 5 as below and still succeeds:

 TEST_P(ExtProcIntegrationTest, RequestMessageNewTimeoutNegativeTestUnspecifiedDirection) {
-  newTimeoutWrongConfigTest(true, 500, envoy::config::core::v3::TrafficDirection::UNSPECIFIED);
+  newTimeoutWrongConfigTest(true, 500, envoy::config::core::v3::TrafficDirection(5));
 }

[2023-03-07 19:14:47.302][16][debug][testing] [./test/integration/fake_upstream.h:167] Sent gRPC message: new_timeout {
  message_timeout {
    nanos: 500000000
  }
  traffic_direction: 5
}

So, it's safer to me to explicitly specify what Envoy allows.

Reply:
Removed traffic_direction from the API.